### PR TITLE
feat: add custom utm parameters to mixpanel

### PR DESCRIPTION
**Pull request by Gauravudia**
## Description of the change

> Resolve INT-908

- Adding the support of sending custom utm parameters to mixpanel.
- Added `buildUtmParams` utility to build utm parameters using `campaign` object.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by mihir-4116 — DISMISSED** [review]

**Review by mihir-4116 — DISMISSED** [review]

**Review by shrouti1507 — APPROVED** [review]

File: src/v0/destinations/mp/util.js [hunk 1]
```diff
@@ -173,6 +173,35 @@ const removeDuplicateMetadata = (mergedBatches) => {
...
+ * const campaign = {
+ *   name: 'summer_sale',
+ *   source: 'newsletter',
+ *   medium: 'email'
+ * };
+ * { utm_campaign: 'summer_sale', utm_source: 'newsletter', utm_medium: 'email' }
+ */
+const buildUtmParams = (campaign) => {
+  const utmParams = {};
+  if (campaign) {
```
Comment (utsabc) [inline src/v0/destinations/mp/util.js:192]: Make sure to add a check if `campaign` is a defined object. The current function will fail if `campaign` is an Array or some other truthy value example:  `valid number, string, boolean true`

**Review by utsabc — CHANGES_REQUESTED** [review]

File: src/v0/destinations/mp/util.js (hunk 1 above)
Comment (Gauravudia) [reply src/v0/destinations/mp/util.js]: added the isObject check

File: src/v0/destinations/mp/util.js [hunk 2]
```diff
@@ -189,7 +190,7 @@ const removeDuplicateMetadata = (mergedBatches) => {
  */
 const buildUtmParams = (campaign) => {
   const utmParams = {};
-  if (campaign) {
+  if (isObject(campaign)) {
```
Comment (utsabc) [inline src/v0/destinations/mp/util.js:193]: Add a `isDefinedNotNull` check as well

Comment (Gauravudia) [reply src/v0/destinations/mp/util.js:193]: Not required. IsObjet will check for null/undefined values also.

Comment (Gauravudia) [reply src/v0/destinations/mp/util.js:193]: Have a test case for this in utils.test.js

**Review by koladilip — APPROVED** [review]

File: src/v0/destinations/mp/util.js (hunk 2 above)
Comment (utsabc) [reply src/v0/destinations/mp/util.js:193]: ok

**Review by utsabc — APPROVED** [review]

**Pull request merged**